ls1intum/Artemis
# [Enhancement] Integrate instructor dashboard information into course dashboard

### Checklist
#### General


#### Server
#### Client
#### Changes affecting Programming Exercises

[ ]* - not needed

### Motivation and Context


Instructor dashboard is about to be deprecated. Before removing it from our codebase, we decided to integrate the elements from the instructor dashboard into the course dashboard. This includes adding the graph and extending the table information. 

### Description
Implemented:
- integrated `ngx-chart` into the codebase, together with adding the graph into the course dashboard
- fix the bug of not providing total assessment lock information for non-exam mode (before the data was not fetched from the database and 0 was returned)
- added the course and tutor rating count on server and in the information table
- minor refactoring to make passing the done/total/tutor data easier
- extended the table information to include the percentage information

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Navigate to Course Dashboard
4. Dashboard should contain the graph, as well as the table with the extended data.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage






### Screenshots
